src/libmongoc/tests/test-mongoc-dns.c

```diff
@@ -213,16 +225,28 @@ _test_dns_maybe_pooled (bson_t *test, bool pooled)
       BSON_ASSERT (
          mongoc_uri_get_tls (_mongoc_client_pool_get_topology (pool)->uri) ==
          expect_ssl);
+#ifdef MONGOC_ENABLE_SSL
+      mongoc_client_pool_set_ssl_opts (pool, &ssl_opts);
+#else
       test_framework_set_pool_ssl_opts (pool);
```

Comment:
The DNS tests require libmongoc to have SSL enabled. `test_framework_set_pool_ssl_opts`  aborts if libmongoc was not built with SSL support. I think you can remove the `#ifdef MONGOC_ENABLE_SSL` in this file.

Reply:
As we discussed in Slack, some of the SSL-related functions are conditionally compiled, while others are always compiled and throw a runtime error if SSL is not enabled.  We will need to retain these conditionals for now.